ReOpen Configuration on Launch

jacobson3 opened the pull request:
Fix #303 and Fix #290

Added INI file to list of standard DCAF folders (along with cache and glyph folders) and created first token which lists the configuration open when the editor exits.

jacobson3 committed:
ReOpen Configuration on Launch

Fix #303 and Fix #290

Added INI file to list of standard DCAF folders (along with cache and glyph folders) and created first token which lists the configuration open when the editor exits.

Commit:
Merge branch 'master' into INI-Configuration

jacobson3 commented:
@becega These changes add an INI file for the DCAF editor.

Other than checking for code quality and completeness I would double check
1. What section names we should be using. I used "DCAF" but I'm not sure if it would make more sense in the long run to have different sections for runtime/edit time.
2. I just created a new folder for INI Token accessors. I imagine this will only grow so it may be worth deciding whether that folder is in the correct location, whether we should have a .lvlib for it, etc.

jacobson3 committed:
Refactor INI API

Moved INI Accessors into a DCAF INI API which relies on storing INI Refnum in FGV instead of having to pass the reference wire everywhere we need to read/write to INI tokens. This should hopefully make it much easier to read/write new tokens in the editor.

jacobson3 committed:
Update source/Standard Editor/Configuration Utility.vi

Changed dataflow in Exit case to prevent race conditions in RTE.

DCAF-Builder commented:
Bleep bloop!

LabVIEW Diff Robot here with some diffs served up hot for your pull request.

Notice something funny? Help fix me on [my GitHub repo.](https://github.com/LabVIEW-DCAF/buildsystem)


### Standard Editor.lvlib--DCAF INI.lvlib--Close DCAF INI.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/13%3A57%3A47/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Close%20DCAF%20INI.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--fgv_DCAF INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/13%3A57%3A47/Standard%20Editor.lvlib--DCAF%20INI.lvlib--fgv_DCAF%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Get INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/13%3A57%3A47/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Get%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Read CurrentPCFG.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/13%3A57%3A47/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Read%20CurrentPCFG.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Set INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/13%3A57%3A47/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Set%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Write CurrentPCFG.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/13%3A57%3A47/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Write%20CurrentPCFG.vi.png)

DCAF-Builder commented:
Bleep bloop!

LabVIEW Diff Robot here with some diffs served up hot for your pull request.

Notice something funny? Help fix me on [my GitHub repo.](https://github.com/LabVIEW-DCAF/buildsystem)


### Standard Editor.lvlib--DCAF INI.lvlib--Close DCAF INI.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/14%3A39%3A53/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Close%20DCAF%20INI.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--fgv_DCAF INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/14%3A39%3A53/Standard%20Editor.lvlib--DCAF%20INI.lvlib--fgv_DCAF%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Get INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/14%3A39%3A53/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Get%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Read CurrentPCFG.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/14%3A39%3A53/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Read%20CurrentPCFG.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Set INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/14%3A39%3A53/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Set%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Write CurrentPCFG.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-19/14%3A39%3A53/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Write%20CurrentPCFG.vi.png)

jacobson3 committed:
Fix Build Crash

StateCluster.ctl got corrupted somehow so I had to replace every instance used by various VIs to stop the build from crashing.

DCAF-Builder commented:
Bleep bloop!

LabVIEW Diff Robot here with some diffs served up hot for your pull request.

Notice something funny? Help fix me on [my GitHub repo.](https://github.com/LabVIEW-DCAF/buildsystem)


### PL Framework Configuration Tool.lvlib--view deploy screen.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/PL%20Framework%20Configuration%20Tool.lvlib--view%20deploy%20screen.vi.png)

### Standard Editor.lvlib--call include vi script.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--call%20include%20vi%20script.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Close DCAF INI.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Close%20DCAF%20INI.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--fgv_DCAF INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--DCAF%20INI.lvlib--fgv_DCAF%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Get INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Get%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Initialize DCAF INI.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Initialize%20DCAF%20INI.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Read CurrentPCFG.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Read%20CurrentPCFG.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Set INI Refnum.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Set%20INI%20Refnum.vi.png)

### Standard Editor.lvlib--DCAF INI.lvlib--Write CurrentPCFG.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--DCAF%20INI.lvlib--Write%20CurrentPCFG.vi.png)

### Standard Editor.lvlib--default file paths.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--default%20file%20paths.vi.png)

### Standard Editor.lvlib--dialog save before deploy.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--dialog%20save%20before%20deploy.vi.png)

### Standard Editor.lvlib--dialog save dirty dot.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--dialog%20save%20dirty%20dot.vi.png)

### Standard Editor.lvlib--Load node UI optimized.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--Load%20node%20UI%20optimized.vi.png)

### Standard Editor.lvlib--Open PCFG.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--Open%20PCFG.vi.png)

### Standard Editor.lvlib--Remove UI without races.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--Remove%20UI%20without%20races.vi.png)

### Standard Editor.lvlib--save as.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--save%20as.vi.png)

### Standard Editor.lvlib--save operation binary.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--save%20operation%20binary.vi.png)

### Standard Editor.lvlib--save operation.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--save%20operation.vi.png)

### Standard Editor.lvlib--Save.vi.png: ![capture](https://raw.githubusercontent.com/DCAF-Builder/diff-pics/master/LabVIEW-DCAF/TagEditorCore/PR-395/2018-10-22/11%3A00%3A11/Standard%20Editor.lvlib--Save.vi.png)

becega reviewed (approved):
Looks Good
